zkevm-circuits/src/rlp_circuit.rs

```diff
@@ -0,0 +1,2368 @@
+                    RlpTxTag::To.expr(),
+                );
+                cb.require_equal(
+                    "tag_index::next == tag_index - 1",
+                    meta.query_advice(rlp_table.tag_index, Rotation::next()),
+                    meta.query_advice(rlp_table.tag_index, Rotation::cur()) - 1.expr(),
+                );
+                cb.require_equal(
+                    "tag_length::next == tag_length",
+                    meta.query_advice(tag_length, Rotation::next()),
+                    meta.query_advice(tag_length, Rotation::cur()),
+                );
+                cb.require_equal(
+                    "value_acc::next == value_acc::cur * 256 + value::next",
+                    meta.query_advice(rlp_table.value_acc, Rotation::next()),
+                    meta.query_advice(rlp_table.value_acc, Rotation::cur()) * 256.expr() +
+                        meta.query_advice(byte_value, Rotation::next()),
+                );
+            });
+
+            // if tag_index == 1
+            cb.condition(is_to(meta) * tindex_eq.clone(), |cb| {
+                cb.require_equal(
+                    "tag::next == RlpTxTag::Value",
+                    meta.query_advice(rlp_table.tag, Rotation::next()),
+                    RlpTxTag::Value.expr(),
+                );
+                cb.require_equal(
+                    "tag_index:next == tag_length::next",
+                    meta.query_advice(rlp_table.tag_index, Rotation::next()),
+                    meta.query_advice(tag_length, Rotation::next()),
+                );
+            });
+
+            //////////////////////////////////////////////////////////////////////////////////////
+            /////////////////////////////////// RlpTxTag::Value //////////////////////////////////
+            //////////////////////////////////////////////////////////////////////////////////////
+            cb.condition(is_value(meta), |cb| {
+                // tag_index < 34
+                cb.require_equal(
+                    "tag_index < 34",
+                    tag_index_lt_34.is_lt(meta, None),
+                    1.expr(),
+                );
+
+                // tag_index >= 1
+                cb.require_zero(
+                    "1 <= tag_index",
+                    not::expr(or::expr([tindex_lt.clone(), tindex_eq.clone()])),
+                );
+            });
+
+            // if tag_index == tag_length && tag_length == 1
+            cb.condition(
+                is_value(meta) * tindex_eq_tlength.clone() * tlength_eq.clone(),
+                |cb| {
+                    cb.require_equal("value < 129", value_lt_129.is_lt(meta, None), 1.expr());
+                    cb.require_equal(
+                        "value == value_acc",
+                        meta.query_advice(byte_value, Rotation::cur()),
+                        meta.query_advice(rlp_table.value_acc, Rotation::cur()),
+                    );
```

Comment:
See the above comments for more details. We need to handle the case `value == 0x80`.